legend-engine-authentication-pure/src/main/resources/core_authentication/metamodel/metamodel.pure

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import meta::pure::authentication::*;
+
+Class <<typemodifiers.abstract>> meta::pure::authentication::AuthenticationSpec
+{
+
+}
+
+Class meta::pure::authentication::OauthAuthenticationSpec extends meta::pure::authentication::AuthenticationSpec
```

Comment:
Lets discuss about the properties

Reply:
1. grantType changed to enum as discussed. 
2. Added scopes